### PR TITLE
Added name property to Command trait

### LINKED ISSUE #3297: Expose useful information about available commands
IntelliJ Scala plugin exposes some information about settings, tasks and commands to the user.

For commands, the API offers only limited ways to give information, especially since the Command trait doesn't have a "name" field. So I hack into it with some reflection and using sbt-internal API:

https://github.com/JetBrains/sbt-structure/blob/master/extractor/src/main/scala-sbt-0.13-1.0/sbt/jetbrains/BadCitizen.scala

used in

https://github.com/JetBrains/sbt-structure/blob/8ba00a43c89eabadfc3af6f54813bf20d8633588/extractor/src/main/scala-sbt-0.13-1.0/org/jetbrains/sbt/extractors/KeysExtractor.scala#L64-L71

A way to access a name for all commands meant for users would make this feature more helpful in general.

### PR DESCRIPTION
**Pull request by jpastuszek**
Fixes #3297

This changes `SimpleCommand` property name from `name` to `_name` so it may brake some hacks that use it. Use added `name: Option[String]` `Command` trait property instead.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @jpastuszek,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=pull/3531/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

File: main-command/src/main/scala/sbt/Command.scala [hunk 1]
```diff
@@ -16,6 +16,7 @@ import sbt.internal.util.{ AttributeKey, AttributeMap, Util }
  * Typically you would resort to a command when you need to do something that's impossible in a regular task.
  */
 sealed trait Command {
+  def name: Option[String]
```
Comment (dwijnand) [inline main-command/src/main/scala/sbt/Command.scala:19]: Let's add this with an extension method defined in the `Command` object. That way we don't have to change SimpleCommand.

**Review by dwijnand — CHANGES_REQUESTED** [review]

File: main-command/src/main/scala/sbt/Command.scala (hunk 1 above)
Comment (jvican) [reply main-command/src/main/scala/sbt/Command.scala:19]: I think this is indeed a superior way to do it 👍

**Comment by jvican** [comment]
@jpastuszek Thank you for doing this, I'm sure @jastice will be happy to see this PR. :tada:
  
@eed3si9n or @dwijnand: would you kindly label this PR as spree please?

**Comment by jastice** [comment]
Cool, thanks!

**Comment by dwijnand** [comment]
Duplicate of #3671

**Pull request closed without merging**